Source/Fuse.Controls.TimePicker/TimePicker.uno

```diff
@@ -0,0 +1,155 @@
+using Uno;
+using Uno.UX;
+using Uno.Compiler.ExportTargetInterop;
+
+using Fuse;
+using Fuse.Controls.Native;
+using Fuse.Scripting;
+
+namespace Fuse.Controls
+{
+	using Native.iOS;
+	using Native.Android;
+
+	interface ITimePickerView
+	{
+		DateTime Value { get; set; }
+		bool Is24HourView { get; set; }
+
+		void PollViewValue();
+	}
+
+	interface ITimePickerHost
+	{
+		void OnValueChanged();
+	}
+
+	public abstract partial class TimePickerBase : Panel, ITimePickerHost
+	{
+		static Selector _valueName = "Value";
+
+		[UXOriginSetter("SetValue")]
+		/**
+			Gets or sets the current time value selected by the `TimePicker`.
+		*/
+		public DateTime Value
+		{
+			get
+			{
+				var tpv = TimePickerView;
+				return tpv != null
+					? tpv.Value
+					: DateTime.UtcNow;
+			}
+			set { SetValue(value, this); }
+		}
+
+		DateTime _outValue = DateTime.UtcNow;
+		DateTime _inValue = DateTime.UtcNow;
+
+		public void SetValue(DateTime value, IPropertyListener origin)
+		{
+			var tpv = TimePickerView;
+			if (tpv != null && tpv.Value != value)
+			{
+				tpv.Value = value;
+				OnValueChanged(origin);
+			}
+		}
+
+		internal void OnValueChanged(IPropertyListener origin)
+		{
+			lock(this)
+				_outValue = Value;
+			OnPropertyChanged(_valueName, origin);
+		}
+
+		void UpdateValue()
+		{
+			lock(this)
+				Value = _inValue;
+		}
+
+		static Selector _is24HourViewName = "Is24HourView";
+
+		[UXOriginSetter("SetIs24HourView")]
+		/**
+			Used to toggle 24-hour or am/pm view. Default is false (am/pm).
+
+			Currently only implemented for Android, as iOS doesn't expose an explicit API for this.
+		*/
+		public bool Is24HourView
+		{
+			get
+			{
+				var tpv = TimePickerView;
+				return tpv != null
+					? tpv.Is24HourView
+					: false;
+			}
+			set { SetIs24HourView(value, this); }
+		}
+
+		bool _outIs24HourView = false;
+		bool _inIs24HourView = false;
+
+		public void SetIs24HourView(bool value, IPropertyListener origin)
+		{
+			var tpv = TimePickerView;
+			if (tpv != null && tpv.Is24HourView != value)
+			{
+				tpv.Is24HourView = value;
+				OnIs24HourViewChanged(origin);
+			}
+		}
+
+		internal void OnIs24HourViewChanged(IPropertyListener origin)
+		{
+			lock (this)
+				_outIs24HourView = Is24HourView;
+			OnPropertyChanged(_is24HourViewName, origin);
+		}
+
+		void UpdateIs24HourView()
+		{
+			lock(this)
+				Is24HourView = _inIs24HourView;
+		}
+
+		ITimePickerView TimePickerView
+		{
+			get { return (ITimePickerView)NativeView; }
+		}
+
+		void PollViewValue()
+		{
+			var tpv = TimePickerView;
+			if (tpv == null)
+				return;
+
+			tpv.PollViewValue();
+		}
+
+		void ITimePickerHost.OnValueChanged()
+		{
+			OnValueChanged(this);
+		}
+
+		protected override void OnRooted()
+		{
+			base.OnRooted();
+			_outValue = Value;
+			_inValue = Value;
+			_outIs24HourView = Is24HourView;
+			_inIs24HourView = Is24HourView;
+
+			UpdateManager.AddAction(PollViewValue);
```

Comment:
Do the underlying controls not emit an event? Polling isn't so nice.

Reply:
The native events are extremely finicky and esp. with the newer (now-default) material appearances, it's quite often they don't fire at all. This was also a problem with the date pickers. Note the comment here: https://github.com/fusetools/fuselibs-public/pull/529/commits/12505751811384f36358f7579f79a3bff9a29715#diff-3927c03f10105bd8d64ff0c6418ea586R26

Reply:
Shouldn't this be resolved on the Native picker itself then, not in the higher level class? The high level control should assume the implementation works, and workarounds are at that lower-level implementation.

This would also give the native control wrapper better control over when precisely it needs to poll for values (such as only while it's active or has focus).

Reply:
Sure, I can move the impl.